File: Programmation réseaux/exo3-2.py
import socket
import sys

if len(sys.argv) < 2:
    sys.exit("Usage: python client.py <chemin_fichier>")

sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
sock.connect(('127.0.0.1', 8080))

with open(sys.argv[1], "rb") as f:
    sock.sendall(f.read())

sock.close()

File: Programmation réseaux/exo3-2.c
#include <stdio.h>
#include <stdlib.h>
#include <sys/socket.h>
#include <netinet/in.h>
#include <arpa/inet.h>
#include <unistd.h>

int main(int argc, char *argv[]) 
{
    if (argc < 2) 
    {
        return 1;
    }

    int sock = socket(AF_INET, SOCK_STREAM, 0);
    struct sockaddr_in serv_addr = {AF_INET, htons(8080)};
    inet_pton(AF_INET, "127.0.0.1", &serv_addr.sin_addr);
    
    connect(sock, (struct sockaddr*)&serv_addr, sizeof(serv_addr));

    FILE *f = fopen(argv[1], "r");
    char buffer[1024];
    int n;
    while ((n = fread(buffer, 1, 1024, f)) > 0) 
    {
        write(sock, buffer, n);
    }

    fclose(f);
    close(sock);
    return 0;
}